pdns/dnsdist.cc

```diff
@@ -1247,15 +1247,27 @@ void* maintThread()
 {
   int interval = 1;
   size_t counter = 0;
+  uint16_t secondsToWaitLog = 0;
 
   for(;;) {
     sleep(interval);
 
     {
       std::lock_guard<std::mutex> lock(g_luamutex);
       auto f =g_lua.readVariable<boost::optional<std::function<void()> > >("maintenance");
-      if(f)
-        (*f)();
+      if(f) {
+        try {
+          (*f)();
+          secondsToWaitLog = 0;
+        }
+        catch(LuaContext::ExecutionErrorException &e) {
+          if (secondsToWaitLog == 0) {
+            infolog("Error during execution of maintenance function: %s", e.what());
+            secondsToWaitLog = 61;
+          }
+          --secondsToWaitLog;
```

Comment:
If `interval` is not guaranteed to be equal to 1 anymore, the decrement won't happen every second except if we alter the code to do so. This would probably be needed to keep the cache cleaning delay consistent anyhow.